### PR TITLE
Tracked addition of federation, sed support in kube DNS

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
[Analytics]()

The kube DNS app recently gained support for federation (whatever that
is), including a new Salt parameter.  This broke the deployAddons.sh script for cluster ubuntu.  The DNS app also gained alternate
templates, intended to be friendly to `sed`.  Fortunately, those do
not demand a federation parameter.

This PR fixes up the `cluster/ubuntu/deployAddons.sh` script to track those changes, by switching to the `sed`-friendly templates.

### CODE REVIEW

**Comment by MikeSpreitzer** [comment]
This is a bug fix, no release note needed.

**Comment by roberthbailey** [comment]
This looks like a cherry pick candidate because otherwise this will be broken on the 1.3 release branch.

**Comment by quinton-hoole** [comment]
@madhusudancs Could you take a look at this please?

File: cluster/ubuntu/deployAddons.sh [hunk 1]
```diff
@@ -41,8 +41,8 @@ function init {
 
 function deploy_dns {
   echo "Deploying DNS on Kubernetes"
-  sed -e "s/{{ pillar\['dns_replicas'\] }}/${DNS_REPLICAS}/g;s/{{ pillar\['dns_domain'\] }}/${DNS_DOMAIN}/g;" "${KUBE_ROOT}/cluster/saltbase/salt/kube-dns/skydns-rc.yaml.in" > skydns-rc.yaml
-  sed -e "s/{{ pillar\['dns_server'\] }}/${DNS_SERVER_IP}/g" "${KUBE_ROOT}/cluster/saltbase/salt/kube-dns/skydns-svc.yaml.in" > skydns-svc.yaml
+  sed -e "s/\\\$DNS_REPLICAS/${DNS_REPLICAS}/g;s/\\\$DNS_DOMAIN/${DNS_DOMAIN}/g;" "${KUBE_ROOT}/cluster/saltbase/salt/kube-dns/skydns-rc.yaml.sed" > skydns-rc.yaml
+  sed -e "s/\\\$DNS_SERVER_IP/${DNS_SERVER_IP}/g" "${KUBE_ROOT}/cluster/saltbase/salt/kube-dns/skydns-svc.yaml.in" > skydns-svc.yaml
```
Comment (madhusudancs) [inline cluster/ubuntu/deployAddons.sh]: Should you also not switch to the .sed script here for consistency?

**Comment by madhusudancs** [comment]
Minor nit, LGTM otherwise.

File: cluster/ubuntu/deployAddons.sh (hunk 1 above)
Comment (roberthbailey) [inline cluster/ubuntu/deployAddons.sh]: I think so. The first time I looked I didn't see the other .sed file but on second look it's right there.

**Comment by girishkalele** [comment]
@MikeSpreitzer 

The skydns-rc.yaml.base file is transformed into the .in and .sed templates so that its easy for changes to the base file to propagate to various formats.

Any changes to the .sed templates might break the Ubuntu case again, so for people making changes to the DNS templates to have visibility into your consumption of this template, one suggestion is to add your own skydns-rc.yaml.ubuntu generated file and transforms2ubuntu.sed script and add a Makefile target. That would make it a visible file in this directory.

**Comment by MikeSpreitzer** [comment]
(1) Oops, I see I forgot to edit the file extension for the second substitution; I will fix that.

(2) I am surprised by the suggestion to change the Makefile to refer specifically to the cluster/ubuntu/deployAddons.sh script.  I did a little search for "skydns-rc.yaml", and found many other files that also seem to know the required set of substitutions.  The beauty of the sed files is that they can be used by multiple scripts.  It would seem tedious and odd to make special template files for each script.  Perhaps instead we should add a comment somewhere reminding developers to do that search for scripts that use the templates?

**Comment by madhusudancs** [comment]
LGTM for now.

**Comment by madhusudancs** [comment]
@k8s-bot test this issue: #IGNORE

**Pull request merged**